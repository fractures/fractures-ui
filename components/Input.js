import React, { Component } from "react"
import cc from "classcat"

const defaultProps = {
	change: () => null,
	placeholder: "",
	type: "text",
	value: ""
}

class Input extends Component {
	render() {
		const isTextarea = this.props.type === "textarea"
		const inputClasses = cc({
			"fr-input": true,
			"fr-input--input px-1": this.props.type !== "textarea",
			"fr-input--textarea p-1": this.props.type === "textarea",
			"fr-btn-primary": this.props.type === "primary"
		})

		const input = (
			<input
				placeholder={ this.props.placeholder }
				onChange={ e => this.props.change(e.target.value) }
				className={ inputClasses }
				type={ this.props.type }
				defaultValue={ this.props.value }
			/>
		)

		const textarea = (
			<textarea
				placeholder={ this.props.placeholder }
				rows={ 6 }
				onChange={ e => this.props.change(e.target.value) }
				className={ inputClasses }
				defaultValue={ this.props.value }
			/>
		)

		return (
			<div className="flex flex-column">
				{this.props.label && <label className="fr-input__label">{this.props.label}</label>}
				{isTextarea ? textarea : input}
				<style jsx global>{`
					.fr-input {
						background-color: var(--shade-0);
						border: 1px solid var(--shade-3);
						border-radius: 0.125rem;
						color: var(--shade-6);
						outline: none;
						transition: box-shadow 100ms ease-in-out;

						font-size: 1rem;
					}

					.fr-input--input {
						line-height: 2.5rem;
					}

					.fr-input--textarea {
						line-height: 1.5rem;
						resize: vertical;
					}

					.fr-input:focus {
						box-shadow: 0 0 0 0.15rem var(--shade-2);
					}

					.fr-input:active {
						border-color: var(--shade-4);
					}

					.fr-input:placeholder {
						color: var(--shade-3);
					}

					.fr-input::-webkit-input-placeholder {
						color: var(--shade-3);
					}

					.fr-input__label {
						color: var(--shade-5);

						font-size: 0.875rem;
						font-weight: 500;
						line-height: 1.75rem;
					}
				`}</style>
			</div>
		)
	}
}

Input.defaultProps = defaultProps

export default Input
