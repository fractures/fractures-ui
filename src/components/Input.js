import cc from "classcat"
import Label from "./shared/Label"
import React from "react"
import PropTypes from "prop-types"

const propTypes = {
	change: PropTypes.func.isRequired,
	placeholder: PropTypes.string,
	type: PropTypes.string,
	value: PropTypes.string
}

const defaultProps = {
	change: () => null,
	placeholder: "",
	type: "text",
	value: ""
}

const Input = props => {
	const isTextarea = props.type === "textarea"
	const inputClasses = cc({
		"fr-input": true,
		"fr-input--input px-1": props.type !== "textarea",
		"fr-input--textarea p-1": props.type === "textarea",
		"fr-btn-primary": props.type === "primary"
	})

	const input = (
		<input
			placeholder={ props.placeholder }
			onChange={ e => props.change(e.target.value) }
			className={ inputClasses }
			type={ props.type }
			defaultValue={ props.value }
		/>
	)

	const textarea = (
		<textarea
			placeholder={ props.placeholder }
			rows={ 6 }
			onChange={ e => props.change(e.target.value) }
			className={ inputClasses }
			defaultValue={ props.value }
		/>
	)

	return (
		<div className="flex flex-column">
			{props.label && (
				<label>
					<Label>{props.label}</Label>
				</label>
			)}
			{isTextarea ? textarea : input}
			<style jsx global>{`
				.fr-input {
					background-color: var(--fr-ground);
					border: 1px solid var(--fr-500);
					border-radius: 0.125rem;
					color: var(--shade-6);
					outline: none;
					transition: box-shadow 100ms ease-in-out;

					font-size: 1rem;
				}

				.fr-input--input {
					line-height: 2.5rem;
				}

				.fr-input--textarea {
					line-height: 1.5rem;
					resize: vertical;
				}

				.fr-input:focus {
					box-shadow: 0 0 0 0.1875rem var(--fr-focus);
				}

				.fr-input:active {
					border-color: var(--fr-700);
				}

				.fr-input:placeholder {
					color: var(--fr-300);
				}

				.fr-input::-webkit-input-placeholder {
					color: var(--fr-300);
				}
			`}</style>
		</div>
	)
}

Input.propTypes = propTypes
Input.defaultProps = defaultProps

export default Input
