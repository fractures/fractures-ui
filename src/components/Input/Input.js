import cc from "classcat"
import Label from "../Label/Label"
import PropTypes from "prop-types"
import React from "react"
import styled from "styled-components"

const propTypes = {
	className: PropTypes.string,
	change: PropTypes.func.isRequired,
	label: PropTypes.string,
	placeholder: PropTypes.string,
	type: PropTypes.string,
	value: PropTypes.string
}

const defaultProps = {
	change: () => null,
	placeholder: "",
	type: "text",
	value: ""
}

const NakedInput = props => {
	const isTextarea = props.type === "textarea"
	const inputClasses = cc({
		"fr-input": true,
		"fr-input--input": props.type !== "textarea",
		"fr-input--textarea": props.type === "textarea",
		"fr-btn-primary": props.type === "primary"
	})

	const input = (
		<input
			placeholder={ props.placeholder }
			onChange={ e => props.change(e.target.value) }
			className={ inputClasses }
			type={ props.type }
			defaultValue={ props.value }
		/>
	)

	const textarea = (
		<textarea
			placeholder={ props.placeholder }
			rows={ 6 }
			onChange={ e => props.change(e.target.value) }
			className={ inputClasses }
			defaultValue={ props.value }
		/>
	)

	return (
		<div className={ props.className }>
			{props.label && (
				<label>
					<Label>{props.label}</Label>
				</label>
			)}
			{isTextarea ? textarea : input}
		</div>
	)
}

const Input = styled(NakedInput)`
	box-sizing: border-box;
	display: flex;
	flex-direction: column;

	.fr-input {
		padding: 0 0.5rem;

		background-color: var(--fr-ground);
		border: 1px solid var(--fr-500);
		border-radius: 0.125rem;
		color: var(--shade-6);
		outline: none;
		transition: box-shadow 100ms ease-in-out;

		font-size: 1rem;
	}

	.fr-input--input {
		line-height: 2.5rem;
	}

	.fr-input--textarea {
		padding: 0.5rem;

		line-height: 1.5rem;
		resize: vertical;
	}

	.fr-input:focus {
		box-shadow: 0 0 0 0.1875rem var(--fr-focus);
	}

	.fr-input:active {
		border-color: var(--fr-700);
	}

	.fr-input:placeholder {
		color: var(--fr-300);
	}

	.fr-input::-webkit-input-placeholder {
		color: var(--fr-300);
	}
`

NakedInput.propTypes = propTypes
NakedInput.defaultProps = defaultProps

Input.propTypes = propTypes
Input.defaultProps = defaultProps

export default Input
