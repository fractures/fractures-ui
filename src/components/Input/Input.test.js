import Input from './Input'

describe('Input', () => {
	const componentRender = <Input />
	const component = renderer.create(componentRender)
	const tree = component.toJSON()

	it('Renders correctly', () => {
		expect(tree).toMatchSnapshot()
	})
})
