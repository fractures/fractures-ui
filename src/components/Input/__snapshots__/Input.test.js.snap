// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders correctly 1`] = `
<div
  className="sc-bwzfXH fGQUIV"
>
  <input
    className="hl-input hl-input--input"
    defaultValue=""
    onChange={[Function]}
    placeholder=""
    type="text"
  />
</div>
`;
